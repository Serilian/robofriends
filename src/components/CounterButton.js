import React, {PureComponent} from 'react';

class CounterButton extends PureComponent {

    constructor(props) {
        super(props);
        this.state = {
            count: 0
        }
    }

    updateCount = () => {
        this.setState((state) => {
             return {count: state.count + 1}
        })
    };

    render() {
        console.log('Counter button');
        return (
            <>
                <button color={this.props.color} onClick={this.updateCount}>Click Me! {this.state.count}</button>
            </>
        );
    }
}

export default CounterButton;