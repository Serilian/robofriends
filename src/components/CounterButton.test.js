import React from 'react';
import {shallow} from 'enzyme';
import CounterButton from './CounterButton';

describe('CounterButton testing', () => {
    it('renders CounterButton component properly', () => {
        const mockColor = 'red';
        expect(shallow(<CounterButton color={mockColor}/>).length).toMatchSnapshot();

    });

    it('corectly increments the counter', () => {
        expect.assertions(2);
        const mockColor = 'red';

        const wrapper = shallow(<CounterButton color={mockColor}/>);

        wrapper.find('[id="counter"]').simulate('click');
        expect(wrapper.state()).toEqual({count: 1});
        wrapper.find('[id="counter"]').simulate('click');
        expect(wrapper.state()).toEqual({count: 2});
    });

});
