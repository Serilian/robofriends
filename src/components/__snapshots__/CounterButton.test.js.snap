// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton testing renders CounterButton component properly 1`] = `1`;
